subt/ros/robot/src/k2.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+
+import math
+import sys
+
+from geometry_msgs.msg import Twist
+import rospy
+import tf
+
+def normalize(quaternion):
+    x0, y0, z0, w0 = quaternion
+    sqr_size = x0*x0 + y0*y0 + z0*z0 + w0*w0
+    if abs(sqr_size - 1.0) > 0.00001:
+        k = math.sqrt(sqr_size)
+        x0, y0, z0, w0 = x0/k, y0/k, z0/k, w0/k
+    return [x0, y0, z0, w0]
+
+
+def euler_zyx(quaternion):
+    x0, y0, z0, w0 = normalize(quaternion)
+    ax =  math.atan2(2*(w0*x0+y0*z0), 1-2*(x0*x0+y0*y0))
+    ay =  math.asin(2*(w0*y0-z0*x0))
+    az =  math.atan2(2*(w0*z0+x0*y0), 1-2*(y0*y0+z0*z0))
+    return [az, ay, ax]
+
```

Comment:
Since we are in ROS, `tf_conversions.transformations.euler_from_quaternion` maybe?

<pre>
>>> help(tf_conversions.transformations.euler_from_quaternion) 

Help on function euler_from_quaternion in module tf.transformations:

euler_from_quaternion(quaternion, axes='sxyz')
    Return Euler angles from quaternion for specified axis sequence.
    
    >>> angles = euler_from_quaternion([0.06146124, 0, 0, 0.99810947])
    >>> numpy.allclose(angles, [0.123, 0, 0])
    True
(END)
</pre>